src/commands/lightning/run/lwc/test.ts

```diff
@@ -20,48 +19,81 @@ export type RunResult = {
+  public static strict = false;
+  public static '--' = true;
   public static readonly flags = {
     debug: Flags.boolean({
       char: 'd',
-      summary: messages.getMessage('debugFlagDescription'),
-      description: messages.getMessage('debugFlagLongDescription'),
+      summary: messages.getMessage('flags.debug.summary'),
+      description: messages.getMessage('flags.debug.description'),
     }),
     watch: Flags.boolean({
-      summary: messages.getMessage('watchFlagDescription'),
-      description: messages.getMessage('watchFlagLongDescription'),
+      summary: messages.getMessage('flags.watch.summary'),
+      description: messages.getMessage('flags.watch.description'),
       exclusive: ['debug'],
     }),
     loglevel,
   };
 
-  // eslint-disable-next-line @typescript-eslint/require-await
   public async run(): Promise<RunResult> {
-    const { args, flags } = await this.parse(RunTest);
+    /* In order to ensure backwards compatibility with old SfdxCommand version of this command
+    * it was necessary to bypass the typical parsing of the command.
+    *
+    * This command defines two flags, debug and watch, which are probably the
+    * most used jest flags. The previous implementation of this command also
+    * defined an args configuration, (passthrough) which would allow any
+    * additional flags to be passed through to jest. This is no longer possible
+    * given how @oclif/core parses the command line, rejecting any flags that
+    * are not defined in the flag configuration.
+    *
+    * The solution is two-fold. First, we change the command to allow non-strict
+    * command configuration, as well as allowing the use of the pass through flag ('--').
+    *
+    * Second, we force the parse to use the config "{strict: false, '--': true}", instead
+    * of passing the command class to the parse method. This allows a parse of the command
+    * as if there are no flags defined, which results in the flags being parsed as arguments.
+    *
+    * Before calling the parse method, we filter out the '--' flag from the this.argv array and then
+    * call the parse method overriding the argv with the first element being '--' and the rest of the
+    * elements being the filtered argv array. This results in the parse method returning the flags
+    * as arguments.
+    *
+    * The resulting argv array is then passed to the runJest method as arguments.
+     */
     const addArgs: string[] = [];
 
-    if (flags.debug) {
-      addArgs.push('--debug');
-    } else if (flags.watch) {
-      addArgs.push('--watch');
-    }
-    if (args.passthrough) {
-      addArgs.push(args.passthrough);
+    // remove the '--', '--json' and 'loglevel' flags from the this.argv array
+    const tArgv = this.argv.filter((arg) => !['--', '--json', 'loglevel'].includes(arg));
+
+    const hasWatchFlag = tArgv.includes('--watch');
+    const hasDebugFlag = tArgv.some(arg => /--debug|-d/.test(arg));
+
```

Comment:
false positive for `--detectopenhandles`.  Regex might need start/end anchors?
https://jestjs.io/docs/cli#--detectopenhandles